lightly_studio/tests/core/test_add_samples.py

```diff
@@ -228,6 +235,112 @@ def test_create_label_map(db_session: Session) -> None:
     assert label_map_2[2] not in label_map_1.values()  # bird is new
 
 
+def test_tag_samples_by_directory_tag_depth_invalid(
+    db_session: Session,
+) -> None:
+    """Tests that tag_depth > 1 raises an error."""
+    # We don't need a full dataset, just the function call
+    with pytest.raises(
+        NotImplementedError,
+        match="tag_depth > 1 is not yet implemented for add_samples_from_path",
+    ):
+        add_samples.tag_samples_by_directory(
+            session=db_session,
+            dataset_id=UUID(int=0),
+            input_path=".",
+            sample_ids=[],
+            tag_depth=2,
+        )
+
+
+def test_tag_samples_by_directory_tag_depth_0(
+    db_session: Session,
+) -> None:
+    """Tests the default behavior (tag_depth=0) adds samples but no tags."""
+    mock_root_path = "/mock/path"
+    mock_image_paths = [
+        f"{mock_root_path}/root_img.png",
+        f"{mock_root_path}/site_1/img1.png",
+    ]
+
+    dataset_table = create_dataset(db_session, "test_dataset")
+
+    images_to_create = [
+        ImageStub(path=mock_image_paths[0]),
+        ImageStub(path=mock_image_paths[1]),
+    ]
+    created_images = create_images(
+        db_session=db_session,
+        dataset_id=dataset_table.dataset_id,
+        images=images_to_create,
+    )
```

Comment:
This can be shortened a lot, no need for all the temporary variables.

```suggestion
    dataset_table = create_dataset(db_session, "test_dataset")
    mock_root_path = "/mock/path"
    created_images = create_images(
        db_session=db_session,
        dataset_id=dataset_table.dataset_id,
        images=[
            ImageStub(path=f"{mock_root_path}/root_img.png"),
            ImageStub(path=f"{mock_root_path}/site_1/img1.png")
        ],
    )
```